pkg/kubelet/container/helpers.go

```diff
@@ -102,6 +104,16 @@ func HashContainer(container *v1.Container) uint64 {
 	hash := fnv.New32a()
 	// Omit nil or empty field when calculating hash value
 	// Please see https://github.com/kubernetes/kubernetes/issues/53644
+	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+		// In-Place Pod Vertical Scaling allows mutable Resources field.
+		// Changes to this field may not require container restart depending
+		// on policy. So it is excluded from hash.
+		cResources := container.Resources
```

Comment:
If we are going to modify the container object, can we at least get a shallow copy of the container object? Just in case there is another goroutine using the container object at the same time. It shouldn't happen today, but just in case.

Reply:
fixed.